c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,101 @@
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        private string _leftJsonString = "";
+        private string _rightJsonString = "";
+
+        public JsonDiff(string left, string right)
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
+            _leftJsonString = left;
+            _rightJsonString = right;
+        }
+
+        public JObject NumericDiff(bool isAbsoluteDiff)
+        {
+            string diff = "";
+            try
             {
-                if (firstKeys.Contains(key))
+                diff = JsonDiffBetweenTwo(_leftJsonString, _rightJsonString);
+                if (diff == null)
                 {
-                    matches.Add(key);
+                    return JObject.Parse(@"{}");
                 }
+                JObject diffJsonObject = JObject.Parse(diff);
+                JObject diffReturnObject = CalculateNumericDiff(diffJsonObject, isAbsoluteDiff);
+                return diffReturnObject;
+            } catch(Exception e)
+            {
+                return JObject.Parse(diff + e.StackTrace);
             }
-            foreach(string key in matches)
+        }
+
+        public string NumericDiffAsString(bool isAbsoluteDiff)
+        {
+            JObject diff = NumericDiff(isAbsoluteDiff);
+            return diff.ToString(Formatting.Indented);
+
+        }
+
+        private JObject CalculateNumericDiff(JObject jObject, bool isAbsoluteDiff)
+        {
+            foreach (KeyValuePair<string, JToken?> prop in jObject)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                if(prop.Key == "_t")
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    continue;
+                }
+                if (prop.Value != null)
+                {
+                    if (prop.Value is JArray)
+                    {
+                        JArray valuesAsArray = (JArray)prop.Value;
+                        if (valuesAsArray[0] != null && valuesAsArray[1] != null)
```

Comment:
Check the length of the array before indexing it.

Reply:
No I think there would be array exceptions when trying to get values outside of the array. I added a check for the length